### ISSUE TITLE
How to avoid Kotlin vulnerabilities in Android Kotlin project with Kotlin-DSL

### ISSUE THREAD

**Issue by Moes81** [dependency-check/dependency-check-gradle#195]
Hello!
Also from my side: Thank you very much, for this awesome project, and all the effort you're putting into it!

To my question:
I have a multi-module Android project (1 app, 5 android-lib modules), written in Kotlin using the Gradle Kotlin-DSL. I'm very unsure, how to configure the dependencyCheck tasks.
"Analyze" will always give me 11 vulnerabilities, all caused by simply using Kotlin as language. "Aggregate" on the other hand, is giving me 0 (zero) vulnerabilities, no matter how low I set `failBuildOnCVSS`. Based on [this SO thread](https://stackoverflow.com/questions/54653400/android-cve-check) skipping `lintClassPath` or adding `releaseCompileClasspath` should fix it. But...well...it's not.

My configuration is as follows:
```
// project build.gradle.kts

subprojects {
    
    // all the other stuff

    apply(plugin = "org.owasp.dependencycheck")
    extensions.findByType(org.owasp.dependencycheck.gradle.extension.DependencyCheckExtension::class)?.let {
        // skipConfigurations added because of: https://stackoverflow.com/questions/54653400/android-cve-check
        // but doesn't help at all. The other suggestion using scanConfigurations doesn't work either.
        it.skipConfigurations = listOf("lintClassPath")
        it.suppressionFile = "${projectDir.absolutePath}/src/main/res/xml/cve_suppressions.xml"
        it.failBuildOnCVSS = 3F //or 9F - it doesn't matter, always getting the same 11 CVEs
    }
}

```
It doesn't matter, if I have this configured in `subprojects {}` or separately in the individual `build.gradle.kts`. 

These are the reported vulnerabilities I get in each module:
```
kotlin-stdlib-jdk8-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk8@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-reflect-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-reflect@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-stdlib-jdk7-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk7@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-stdlib-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-stdlib-common-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-common@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-android-extensions-runtime-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-android-extensions-runtime@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-compiler-embeddable-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-compiler-embeddable@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-script-runtime-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-script-runtime@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-daemon-embeddable-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-daemon-embeddable@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-android-extensions-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-android-extensions@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
kotlin-klib-commonizer-embeddable-1.4.0.jar (pkg:maven/org.jetbrains.kotlin/kotlin-klib-commonizer-embeddable@1.4.0, cpe:2.3:a:jetbrains:kotlin:1.4.0:milestone1:*:*:*:*:*:*) : CVE-2020-15824
```

I'd really appreciate some help to figure out the right configuration here.

Edit:
I using the following versions:
- DependencyCheck Plugin: 5.3.2
- Kotlin: 1.4.0
- Gradle Build Tools: 4.0.1
- Gradle 6.1.1

**Comment by alexburlton-sonocent** [comment]
Yeah this looks like a false positive in the check to me - I'm also seeing failures due to CVE-2020-15824 despite being on kotlin version 1.4.10. Tried bumping to the latest version of the dependencyCheck (6.0.1) and it's still happening there.

Per the CVE:

> In JetBrains Kotlin from 1.4-M1 to 1.4-RC (as Kotlin 1.3.7x is not affected by the issue. Fixed version is 1.4.0)

So the check should not be flagging `1.4.0` or `1.4.10` as vulnerable.

**Comment by triplem** [comment]
Still happening with 1.4.21-2.

CVE-2020-15824, CVE-2020-29582, whereas the latter one is fixed in 1.4.21, according to NVD

**Comment by jeremylong** [comment]
You can create a local suppression file or even submit a PR to add the suppression rules.
